### PR TITLE
CFIN-405 add synonyms from course finder

### PR DESCRIPTION
**Pull request by RichardAllum**
I couldn't find a syntax to map multiple words to the same result without using regex, which was avoided here as we don't know that the people doing the maintenance of this will know it.

This file might not want to be here as it's only edited through the GUI also (cannot be edited through the administer files screen).

### CODE REVIEW

**Comment by ce666-york** [comment]
Are we able to deploy these just to preview, so we can see a before/after? both dev/prod seem to be returning the same results for me for the synonyms I've tried.

If not, perhaps use a temp collection to do so?

File: config/synonyms.cfg [hunk 1]
```diff
@@ -0,0 +1,20 @@
+PADRE Thesaurus Version: 3
++=sps='social and political studies'=v2=3908a623-23a4-4620-ba2b-ac161127c98c=A20201110093743.514Z=A20201110093743.514Z
++=tftv='theatre film television'=v2=878abd41-d262-4896-9609-eaaa54962d0d=A20201110093743.514Z=A20201110093743.514Z
++=pep='philosophy economics politics'=v2=bea188d5-cb6a-4f20-a510-95b81cb37c36=A20201110093743.515Z=A20201110093743.515Z
++=tv='television'=v2=c1297e0f-5deb-4829-929d-95fa2927f4ad=A20201110093743.520Z=A20201110093743.520Z
++=ppe='philosophy politics economics'=v2=5f334fa4-e1cd-45f3-a382-9f4c805586d9=A20201110093743.520Z=A20201110093743.520Z
++=spsw='social policy social work'=v2=2cf79412-fd6d-4bef-89ab-ac157fef4a43=A20201110093743.521Z=A20201110093743.521Z
++=medical='medicine'=v2=36f3430a-0fe8-4db2-9815-e20b12b23af5=A20201110093743.547Z=A20201110093743.547Z
++=performing='writing'=v2=5fca92d7-9ab9-4094-85aa-231fd904232a=A20201110093743.547Z=A20201110093743.547Z
```
Comment (ce666-york) [inline config/synonyms.cfg:9]: Some of these are odd, for example it feels strange that searching for `performing` wouldn't actually include any results for performing, just `writing`. This is what the Course Finder config says though.

You can see that in a search for `performing` with synonyms:

https://york.funnelback.co.uk/search?query=performing&collection=york-uni-courses&profile=_default_preview&form=course-search&smeta_contentType=course&num_ranks=5

"Writing, Directing and Performance" is at the top, which is good, but then for a search without synonyms:

https://york.funnelback.co.uk/search?query=performing&collection=york-uni-courses&profile=_default_preview&form=course-search&smeta_contentType=course&num_ranks=5&synonyms=off

The top result is "Mathematics and Physics", which isn't great, but in position 3 is "Music" which does seem related to performing. This isn't a great example, but I wonder if we're creating some sub-optimal situations by not doing more mappings like `performing='[performing, writing]'`.

In Course Finder [`performing` just returns the one result now](https://www.york.ac.uk/study/undergraduate/courses/search/?mode=&q=performing&level=undergraduate), so I don't know how good or bad the results were before.

I think it's worth having a more critical eye on these synonyms.

**Comment by ce666-york** [comment]
I've just seen in [the Funnelback documentation](https://docs.funnelback.com/customise/standard-options/synonyms.html) that you can perform a search without synonyms by including the `synonyms=off` as a query parameter.

File: config/synonyms.cfg (hunk 1 above)
Comment (FergusMcGlynn) [reply config/synonyms.cfg:9]: `acting='writing'` is another odd looking entry, like `performing='writing'`. I'm guessing these were added to Course Finder to bump "Writing, Directing and Performance" to the top for these search terms.

**Comment by FergusMcGlynn** [comment]
> Are we able to deploy these just to preview, so we can see a before/after? both dev/prod seem to be returning the same results for me for the synonyms I've tried.
> 
> If not, perhaps use a temp collection to do so?

If we can't use `_default_preview` for viewing synonym changes, and have to use another collection, then it might be worth amending the Courses API so that the `collection` parameter it uses to make Funnelback requests can be set as an environment variables. Then we could make the Courses API use different collections in dev and production, to make user testing of changes a bit simpler. I'm not saying that you would need to do this work, btw. We could create a new sub-task for it under this parent story and anyone could implement it.

File: config/synonyms.cfg (hunk 1 above)
Comment (ncn1) [reply config/synonyms.cfg:9]: Perhaps the synonym for performing would best be 'performance' - a search for performance brings up 'Theatre: Writing, Directing and Performance' first, then Music, then Musi and sound recording

**Comment by RichardAllum** [comment]
It should be possible to use default_preview for checking synonym changes, changes made in the marketing dashboard screen are immediately applied to the live search but to not appear to be applied to the preview search immediately. As you discovered, it's also possible to turn off synonyms with a query parameter.

I think it may be worth adding the ability for the course search front-end to turn off synonyms by passing that query parameter through to funnelback, but I'm not sure that we want to do anything to set up an entire synonym-previewing environment. If you have access to the marketing console, it is possible to edit/publish/unpublish individual synonyms with one click and they will instantly be applied to the live search, without any intermediary steps.

**Pull request closed without merging**